### ISSUE TITLE
Problems with dimensions without coordinates

### ISSUE THREAD

**Issue by jbusecke** [rabernat/xrft#17]
Hey, 
I just ran into a problem with an array that has dimensions w/o coordinates. If I run this little example:
```
import xrft
import xarray as xr
import numpy as np

data = xr.DataArray(np.random.random([20,30,100]),dims=['time','lat','lon'])
ps = xrft.power_spectrum(data,dim=['time'])
```

It gives a key error:
```
KeyErrorTraceback (most recent call last)
<ipython-input-17-236267e01ceb> in <module>()
      6 data = xr.DataArray(np.random.random([20,30,100]),dims=['time','lat','lon'])
      7 print data
----> 8 ps = xrft.power_spectrum(data,dim=['time'])
      9 
     10 # # But that is not the main issue here

/home/julius/code/conda_envs/src/xrft/xrft/xrft.pyc in power_spectrum(da, dim, shift, detrend, density, window)
    273     daft = dft(da,
    274             dim=dim, shift=shift, detrend=detrend,
--> 275             window=window)
    276 
    277     coord = list(daft.coords)

/home/julius/code/conda_envs/src/xrft/xrft/xrft.pyc in dft(da, dim, shift, detrend, window)
    219             newcoords[d] = da.coords[d].values
    220         else:
--> 221             newcoords[d] = k_coords[d]
    222 
    223     dk = [l[1] - l[0] for l in k]

KeyError: 'lat'
```

If I run the same example with coordinates it works fine:

```
import xrft
import xarray as xr
import numpy as np

data = xr.DataArray(np.random.random([20,30,100]),dims=['time','lat','lon'],coords={'time':range(20),'lat':range(30),'lon':range(100)})
ps = xrft.power_spectrum(data,dim=['time'])

```

Best Julius

**Comment by roxyboy** [comment]
I think we would like to keep this feature of giving an error when data without coordinates is given as an input. There wouldn't be the point for `xrft` if the data didn't have metadata and the coordinates are used to check whether the data is evenly spatially spaced.

**Comment by jbusecke** [comment]
I see. That makes sense. But perhaps it qould be good to give a more verbose error message if that particular line throws a key error. Or maybe check beforehand if there are dimensions without coordinates?

**Comment by roxyboy** [comment]
I'm not sure how to create an `if` syntax to check whether the `xarray.DataArray` has no coordinates?

**Comment by rabernat** [comment]
> I think we should keep this feature of giving an error when data without coordinates is given as an input.

I disagree. We should support all forms of xarray data. The desired behavior is simple: If the input doesn't have coordinates, then neither should the output.

**Comment by roxyboy** [comment]
I'm not sure how to check whether coordinates are given in a smart fashion. Even if coordinates are not given when the data is generated, `xarray.DataArray` gives a default `np.arange()` coordinate to `dim_0`

**Comment by rabernat** [comment]
I already fixed this and tested for it in #20. Please review that PR to see how it works.

**Comment by roxyboy** [comment]
Thank you very much. It makes sense. Could you merge both pull requests?

Best,
Takaya
